lynx/common/actions/chop.py

```diff
@@ -0,0 +1,29 @@
+from dataclasses import dataclass
+
+from lynx.common.actions.action import Action
+from lynx.common.vector import Vector
+from lynx.common.actions.removeObject import RemoveObject
+from lynx.common.actions.createObject import CreateObject
+from lynx.common.object import Object
+from lynx.common.actions.common_requirements import CommonRequirements
+
+@dataclass
+class Chop(Action):
+	"""
+	Simple action used to hit/destroy object, which stands on the given destination.
+	"""
+	object_id: int = -1
+	position: Vector = Vector(1, 0)
+
+	def apply(self, scene: 'Scene') -> None:
+		objects_on_square = scene.get_objects_by_position(self.position)
+		for object in objects_on_square:
+			if object.name == 'tree':
+				remove_action = RemoveObject(object.id)
+				log = Object(id=scene.generate_id(), name='log', position=object.position, pickable=True)
+				create_action = CreateObject(log.serialize())
+				scene.add_to_pending_actions(remove_action.serialize())
+				scene.add_to_pending_actions(create_action.serialize())
+
+	def satisfies_requirements(self, scene: 'Scene') -> bool:

```

Comment:
Adapt `satisfies_requirements` of `Action`, so it takes two arguments just like this override.
`def satisfies_requirements(self, scene: Optional['Scene'] = None) -> bool:`

Reply:
What's the point? Why would you want to check requirements on a `Scene` that is `None`?

Reply:
I would not want to.
The point is that we should not change method arguments when we override a given method and some `satisfies_requirement` overrides do not take `Scene` argument.